crates/db_views_actor/src/community_view.rs

```diff
@@ -185,35 +186,39 @@ impl CommunityView {
       .await
   }
 
-  pub async fn is_mod_or_admin(
+  pub async fn check_is_mod_or_admin(
     pool: &mut DbPool<'_>,
     person_id: PersonId,
     community_id: CommunityId,
-  ) -> Result<bool, Error> {
+  ) -> LemmyResult<()> {
     let is_mod =
-      CommunityModeratorView::is_community_moderator(pool, community_id, person_id).await?;
-    if is_mod {
-      Ok(true)
-    } else if let Ok(Some(person_view)) = PersonView::read(pool, person_id).await {
-      Ok(person_view.is_admin)
+      CommunityModeratorView::check_is_community_moderator(pool, community_id, person_id).await;
+    if is_mod.is_ok()
+      || PersonView::read(pool, person_id)
+        .await
+        .is_ok_and(|t| t.is_some_and(|t| t.is_admin))
+    {
+      Ok(())
     } else {
-      Ok(false)
+      Err(LemmyErrorType::NotAModOrAdmin.into())
     }
   }
 
   /// Checks if a person is an admin, or moderator of any community.
-  pub async fn is_mod_of_any_or_admin(
+  pub async fn check_is_mod_of_any_or_admin(
     pool: &mut DbPool<'_>,
     person_id: PersonId,
-  ) -> Result<bool, Error> {
+  ) -> LemmyResult<()> {
     let is_mod_of_any =
-      CommunityModeratorView::is_community_moderator_of_any(pool, person_id).await?;
-    if is_mod_of_any {
-      Ok(true)
-    } else if let Ok(Some(person_view)) = PersonView::read(pool, person_id).await {
-      Ok(person_view.is_admin)
+      CommunityModeratorView::is_community_moderator_of_any(pool, person_id).await;
+    if is_mod_of_any.is_ok()
+      || PersonView::read(pool, person_id)
+        .await
+        .is_ok_and(|t| t.is_some())
+    {
+      Ok(())
     } else {
-      Ok(false)
+      Err(LemmyErrorType::NotAModOrAdmin.into())
```

Comment:
Both of these functions need to properly handle different error types from the moderator check, maybe by directly returning the result at the end like in the functions you already fixed. Also, you wrongly removed the check of `person_view.is_admin` in the 2nd function.

The already existing code also lacks proper error handling for `PersonView::read`, and you can mostly fix that by using `?`. Handling the `Ok(None)` case is not very important.

Reply:
Thx, I fixed adding back in the admin check in the second function. 

These were helper functions to check both at the same time, so if it needs to return 3 options as opposed to a simple pass/fail, then I'd rather just get rid of these ones entirely.

> The already existing code also lacks proper error handling for PersonView::read, and you can mostly fix that by using ?. Handling the Ok(None) case is not very important.

Seems like the `is_ok_and` already handles that case.